artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/openmbean/MessageOpenTypeFactory.java

```diff
@@ -123,6 +124,7 @@ protected void init() throws OpenDataException {
    public Map<String, Object> getFields(M m, int valueSizeLimit, int deliveryCount) throws OpenDataException {
       Map<String, Object> rc = new HashMap<>();
       rc.put(CompositeDataConstants.MESSAGE_ID, "" + m.getMessageID());
+      rc.put(CompositeDataConstants.PROTOCOL, m.getClass().getSimpleName());
```

Comment:
This is a bit dangerous as it exposes class name to api, can we make this something more explicit e.g. add interface to message api such as (getProtocol) and then for the protocol implementation classes to implement that returning a constant value.

Reply:
That was my original thought, until I saw how self-descriptive the class-names were.
I'll follow the original thought and your suggestion.
Still, I'll make the default implementation getClass().getSimpleName() because there are internal classes and client classes for which the name is never shown.

Reply:
please ignore the remark about getClass().getSimpleName(). there is only one internal class that is not a sub of CoreMessage. so that problem is negligable.